extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java

```diff
@@ -166,15 +175,21 @@ private void applyAssumeRole(
       AWSCredentialsProvider awsCredentialsProvider
   )
   {
-    String assumeRoleArn = s3InputSourceConfig.getAssumeRoleArn();
-    if (assumeRoleArn != null) {
+    // Do not run if WebIdentityToken file and assumeRole ARN are detected from the environment variable,
+    // we want the default s3ClientBuilder behavior for ServiceAccount + eks.amazonaws.com/role-arn annotation to work.
```

Comment:
Can you please tell us how did you arrive at the above conclusion. Link to the relevant AWS documentation would be helpful.

Reply:
This is the default credentials provider which is turned on by default on S3 Client (provided as reference). You can see that the WebIdentityToken is in the middle (enabled when `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_ARN` are configured in environment variables) https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/DefaultCredentialsProvider.html

I will give you a little background:

Everyone who runs Druid on EKS wants to use the ServiceAccount with annotated Role ARN because it's more secure and uses AWS's newer IMDS v2 API.

When we actually configured "ServiceAccount with annotated Role ARN", `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_ARN` will be configured inside the pod's environment variables.

Now, the old codebase somehow picked up the `AWS_ROLE_ARN` from the environment variable and populate the S3InputSourceConfig.assumeRole field. And that caused the code to execute that special assumeRole function. We don't want that to happen.

Reply:
Based on reading: https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html IMHO `AWS_WEB_IDENTITY_TOKEN_FILE` should be the lowest priority of authentication that we should support as it looks like its more supported for short duration access to AWS services. 
However, I would somehow first check why AWS_ROLE_ARN got picked up. Are you specifying it in the ingestion spec somewhere?

Reply:
@cryptoe `I would somehow first check why AWS_ROLE_ARN got picked up. Are you specifying it in the ingestion spec somewhere?`

I am surprised about this too, I did not specify it anywhere inside the ingestion spec. Why did the S3InputConfig object picked it up? I could not figure that out.